secure/storage/src/on_disk.rs

```diff
@@ -28,9 +28,14 @@ impl OnDiskStorage {
             File::create(&file_path).expect("Unable to create storage");
         }
 
+        // If there is no path just a file use the same.
+        let file_dir = file_path
+            .parent()
+            .map_or(PathBuf::new(), |p| p.to_path_buf());
+
```

Comment:
Curious why we should support this behavior?  Seems like a special case we shouldn't expect?

Reply:
Improved comment.